File: VB/HowToImplementSingleton.Web/WebApplication.vb
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Xpo
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports DevExpress.ExpressApp.Web

Namespace HowToImplementSingleton.Web
    Partial Public Class HowToImplementSingletonAspNetApplication
        Inherits WebApplication

        Protected Overrides Sub CreateDefaultObjectSpaceProvider(ByVal args As CreateCustomObjectSpaceProviderEventArgs)
            args.ObjectSpaceProvider = New XPObjectSpaceProvider(args.ConnectionString, args.Connection, True)
        End Sub
        Private module1 As DevExpress.ExpressApp.SystemModule.SystemModule
        Private module2 As DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule
        Private howToImplementSingletonModule1 As HowToImplementSingleton.Module.HowToImplementSingletonModule
        Private module3 As HowToImplementSingleton.Module.HowToImplementSingletonModule
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub HowToImplementSingletonAspNetApplication_DatabaseVersionMismatch(ByVal sender As Object, ByVal e As DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs) Handles MyBase.DatabaseVersionMismatch
            e.Updater.Update()
            e.Handled = True
        End Sub

        Private Sub InitializeComponent()
            Me.module1 = New DevExpress.ExpressApp.SystemModule.SystemModule()
            Me.module2 = New DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule()
            Me.module3 = New HowToImplementSingleton.Module.HowToImplementSingletonModule()
            Me.howToImplementSingletonModule1 = New HowToImplementSingleton.Module.HowToImplementSingletonModule()
            DirectCast(Me, System.ComponentModel.ISupportInitialize).BeginInit()
            ' 
            ' HowToImplementSingletonAspNetApplication
            ' 
            Me.ApplicationName = "HowToImplementSingleton"
            Me.Modules.Add(Me.module1)
            Me.Modules.Add(Me.module2)
            Me.Modules.Add(Me.howToImplementSingletonModule1)
            DirectCast(Me, System.ComponentModel.ISupportInitialize).EndInit()

        End Sub
    End Class
End Namespace

File: CS/HowToImplementSingleton.Web/WebApplication.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Xpo;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using DevExpress.ExpressApp.Web;

namespace HowToImplementSingleton.Web {
	public partial class HowToImplementSingletonAspNetApplication : WebApplication {
        protected override void CreateDefaultObjectSpaceProvider(CreateCustomObjectSpaceProviderEventArgs args) {
            args.ObjectSpaceProvider = new XPObjectSpaceProvider(args.ConnectionString, args.Connection, true);
        }
		private DevExpress.ExpressApp.SystemModule.SystemModule module1;
		private DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule module2;
        private HowToImplementSingleton.Module.HowToImplementSingletonModule howToImplementSingletonModule1;
		private HowToImplementSingleton.Module.HowToImplementSingletonModule module3;
        public HowToImplementSingletonAspNetApplication() {
			InitializeComponent();
		}

		private void HowToImplementSingletonAspNetApplication_DatabaseVersionMismatch(object sender, DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs e) {
		    e.Updater.Update();
			e.Handled = true;
		}

		private void InitializeComponent() {
            this.module1 = new DevExpress.ExpressApp.SystemModule.SystemModule();
            this.module2 = new DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule();
            this.module3 = new HowToImplementSingleton.Module.HowToImplementSingletonModule();
            this.howToImplementSingletonModule1 = new HowToImplementSingleton.Module.HowToImplementSingletonModule();
            ((System.ComponentModel.ISupportInitialize)(this)).BeginInit();
            // 
            // HowToImplementSingletonAspNetApplication
            // 
            this.ApplicationName = "HowToImplementSingleton";
            this.Modules.Add(this.module1);
            this.Modules.Add(this.module2);
            this.Modules.Add(this.howToImplementSingletonModule1);
            this.DatabaseVersionMismatch += new System.EventHandler<DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs>(this.HowToImplementSingletonAspNetApplication_DatabaseVersionMismatch);
            ((System.ComponentModel.ISupportInitialize)(this)).EndInit();

		}
	}
}
